index.js

```diff
@@ -1,15 +1,22 @@
 import React from 'react';
+import PropTypes from 'prop-types';
+import createButton from '@aeternity/superhero-button';
 
-import { linkToExtension, genTipDeeplink } from '@aeternity/superhero-button_fork';
+export default class Button extends React.Component {
+  constructor(props) {
+    super(props);
+    this.button = React.createRef();
+    this.componentDidMount = this.componentDidUpdate = () =>
+      createButton(this.button.current, this.props);
```

Comment:
@davidyuk I'm aftraid of that situation:
- every `componentDidUpdate ` we creat a `createButton` scrope
- we fill the memory with a lot of `createButton` instances
- GC can't handle that

Also I think that `this.componentDidUpdate` need some prop that check like [prevProps !== newProps](https://github.com/aeternity/jitsi-meet/blob/90fa02b4e1533923bed657cb83e6765fae14aeaa/react/features/toolbox/components/web/Toolbox.js#L340-L342).

Will the scope be removed after the componentDidMount?

Reply:
@davidyuk also I'm afraid that every `componentDidUpdate` will be create a new instance for the button with `new score`.

Reply:
that's what I'm afraid of, that's the point – memory

Reply:
Well, `createButton` is not a constructor, it does not create any instances, it just replaces a DOM node with a button. Why do you think it has something that should be GC?

Reply:
but it also have some cope that fills the memory.

Reply:
and I'm afraid that it's no matter constructor it or not.